### DISCUSSION TITLE
Permissions work in local but not on the server.

### DISCUSSION THREAD

**Discussion by jesusprofe** [spatie/laravel-permission#2644] [Q&A] [answered]
Hello.
I have an app running in local (in mac with php artisan serve) but when I deploy it in a server with apache/debian the roles don’t work. 
Directives for blade like @can works like a charm in local but in debian it’s like there are no roles.
If I use dd(Auth::user()->getRoleNames()); in the controller, I get correct names...
What am I missing?

**Comment by drbyte** [comment, marked as answer]
`@can` uses Laravel's `auth` middleware to check whether the user has the specified `permission`. Doesn't check roles, only permissions assigned TO those roles.

You said `getRoleNames()` reports roles for the user. Great. But do those roles have any permissions assigned to them? And do they match the guard_name that the user is logged in via?
